src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.netcoreapp.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.Hosting.Internal
+{
+    public partial class ConsoleLifetime : IHostLifetime
+    {
+        private PosixSignalRegistration _sigTermRegistration;
+
+        private partial void RegisterShutdownHandlers()
+        {
+            _sigTermRegistration = PosixSignalRegistration.Create(PosixSignal.SIGTERM, context => OnSigTerm(context));
```

Comment:
Do we expect this to be called more than once?  It could just be:
```suggestion
            _sigTermRegistration = PosixSignalRegistration.Create(PosixSignal.SIGTERM, OnSigTerm);
```

Reply:
Actually, doesn't matter if it's called more than once: OnSigTerm is an instance method, so this is going to allocate a delegate regardless.

Reply:
> Do we expect this to be called more than once?

No, typically there is only one instance of a Host per app. So we don't expect it to be called more than once.